ui/app/services/flags.ts

```diff
@@ -24,22 +24,37 @@ export default class flagsService extends Service {
   @service declare readonly version: VersionService;
   @service declare readonly store: StoreService;
 
-  @tracked flags: string[] = [];
   @tracked activatedFlags: string[] = [];
+  @tracked featureFlags: string[] = [];
 
-  setFeatureFlags(flags: string[]) {
-    this.flags = flags;
+  get isManaged(): boolean {
+    return this.featureFlags.includes(FLAGS.vaultCloudNamespace);
   }
 
-  get managedNamespaceRoot() {
-    if (this.flags && this.flags.includes(FLAGS.vaultCloudNamespace)) {
-      return 'admin';
+  get managedNamespaceRoot(): string | null {
+    return this.isManaged ? 'admin' : null;
+  }
+
+  getFeatureFlags = keepLatestTask(async () => {
```

Comment:
Note:  I moved the fetch from the `beforeModel` hook in the application route to this concurrency task. It follows the pattern of `getActivatedFeatures`. However, there is a small difference in my pattern and I wanted to point that out. Compare the try/catch here to the previous call:
```
// previous call
const result = await fetch('/v1/sys/internal/ui/feature-flags', {
      method: 'GET',
    });
    if (result.status === 200) {
      const body = await result.json();
      const flags = body.feature_flags || [];
      this.flagsService.setFeatureFlags(flags);
    }
```

Reply:
i considered suggesting this change earlier so i'm glad you brought this up. i think it keeps the service's scope nicely organized. i like that this ensures we don't have to manually call `this.flags.setFeatureFlags` outside of the context of the service. 😀